fury/primitive.py

```diff
@@ -851,3 +851,235 @@ def prim_cylinder(radius=0.5, height=1, sectors=36, capped=True):
+    vertices = np.ascontiguousarray(xyz)
+    return vertices, triangles
+
+
+def prim_para_boys_surface():
+    """Return vertices and triangle for Boy's surface
+
+    Returns
+    -------
+    vertices: ndarray
+        vertices coords that compose our Boy's surface
+    triangles: ndarray
+        triangles that compose our Boy's surface
+    """
+
+    npoints = 100
+    sin = np.sin
+    cos = np.cos
+    u = np.linspace(-np.pi/2, np.pi/2, npoints)
+    v = np.linspace(0, np.pi, npoints)
+    u, v = np.meshgrid(u, v)
+    u = u.reshape(-1)
+    v = v.reshape(-1)
+    points2D = np.vstack([u, v]).T
+    tri = Delaunay(points2D)
+    triangles = tri.simplices
+
+    x = (2**0.5*cos(v)**2*cos(2*u) + cos(u)*sin(2*v)) / \
+        (2 - 2**0.5*sin(3*u)*sin(2*v))
+    y = (2**0.5*cos(v)**2*sin(2*u) - sin(u)*sin(2*v)) / \
+        (2 - 2**0.5*sin(3*u)*sin(2*v))
+    z = 3*cos(v)**2 / (2 - 2**0.5*sin(3*u)*sin(2*v))
+    xyz = np.vstack([x, y, z]).T
+    vertices = np.ascontiguousarray(xyz)
+    return vertices, triangles
+
+
+def prim_para_bohemian_dome():
+    """Return vertices and triangle for Bohemian dome
+
+    Returns
+    -------
+    vertices: ndarray
+        vertices coords that compose our Bohemian dome
+    triangles: ndarray
+        triangles that compose our Bohemian dome
+    """
+
+    npoints = 100
+    sin = np.sin
+    cos = np.cos
+    u = np.linspace(0, 2*np.pi, npoints)
+    v = np.linspace(0, 2*np.pi, npoints)
+    u, v = np.meshgrid(u, v)
+    u = u.reshape(-1)
+    v = v.reshape(-1)
+    points2D = np.vstack([u, v]).T
+    tri = Delaunay(points2D)
+    triangles = tri.simplices
+
+    a = 0.5
+    b = 1.5
+    c = 1
+    x = a*cos(u)
+    y = (b*cos(v) + a*sin(u))
+    z = c*sin(v)
+    xyz = np.vstack([x, y, z]).T
+    vertices = np.ascontiguousarray(xyz)
+    return vertices, triangles
+
+
+def prim_para_dinis_surface():
+    """Return vertices and triangle for Dini's surface
+
+    Returns
+    -------
+    vertices: ndarray
+        vertices coords that compose our Dini's surface
+    triangles: ndarray
+        triangles that compose our Dini's surface
+    """
+
+    npoints = 100
+    sin = np.sin
+    cos = np.cos
+    u = np.linspace(0, 4*np.pi, npoints)
+    v = np.linspace(0.01, 1, npoints)
+    u, v = np.meshgrid(u, v)
+    u = u.reshape(-1)
+    v = v.reshape(-1)
+    points2D = np.vstack([u, v]).T
+    tri = Delaunay(points2D)
+    triangles = tri.simplices
+
+    a = 1
+    b = 0.2
+    x = a*cos(u)*sin(v)
+    y = a*sin(u)*sin(v)
+    z = a*(cos(v) + np.log(np.tan(v/2))) + b*u
+    xyz = np.vstack([x, y, z]).T
+    vertices = np.ascontiguousarray(xyz)
+    return vertices, triangles
+
+
+def prim_para_pluckers_conoid(num_folds=2):
+    """Return vertices and triangle for Plücker's conoid
+
+    Returns
+    -------
+    vertices: ndarray
+        vertices coords that compose our Plücker's conoid
+    triangles: ndarray
+        triangles that compose our Plücker's conoid
+    """
+
+    npoints = 100
```

Comment:
same comment as above: can you put it as a function parameter with `100`as default value

Reply:
done